Stereo Effect

anhr opened the pull request:
Now user can choice a spatial multiplex format and some stereo effect
settings. The following spatial multiplex formats are currently
available: Side by side, Top and bottom. See
https://en.wikipedia.org/wiki/DVB_3D-TV for details

anhr committed:
Stereo Effect

Now user can choice a spatial multiplex format and some stereo effect
settings. The following spatial multiplex formats are currently
available: Side by side, Top and bottom. See
https://en.wikipedia.org/wiki/DVB_3D-TV for details

mrdoob commented:
Looks good to me!
Do you mind cleaning the file so it only uses tabs for indentation though?

anhr committed:
bug of loading of spatialMultiplex

anhr committed:
Cosmetic changes

anhr committed:
optimized code

anhr committed:
optimized code

anhr committed:
use tabs for indentation

anhr committed:
use tabs for indentation

anhr committed:
cosmetic changes

anhr committed:
StereoEffects gui folder

Some developers do not want use dat-gui library for StereoEffect user
settings. I have moved adding StereoEffects folder into dat-gui from
StereoEffects to another file located in new examples\js\gui folder.
This file saved as UTF-8 for localization.

anhr committed:
Resolving of github commit problems

anhr committed:
Resolving of github commit problems.

anhr committed:
Resolving of github commit problems.

anhr committed:
Merge remote-tracking branch 'refs/remotes/mrdoob/dev' into dev

Mugen87 commented on src/objects/SpriteText.js:
```diff
@@ -0,0 +1,142 @@
+//	}
+//}
+//Thanks to / https://github.com/vasturiano/three-spritetext
+function SpriteText( options ) {
```
I don't think this file should be in the core. Please put it in your example.

anhr replied:
Now I have removed SpriteText.js because this file from my future pull request. I included this file in mistake. Sorry. See my commit https://github.com/mrdoob/three.js/pull/15434/commits/57876357c9efbf5de96a514312b49a91da0236aa

Mugen87 commented:
I don't think it's good that this PR creates a lot of new files (e.g. `examples/js/gui/StereoEffect.js`, `examples/js/gui/gui.js` or `src/objects/SpriteText.js`). I would prefer modifications to `StereoEffect` and the respective example.

Mugen87 commented on .gitignore:
```diff
@@ -8,3 +8,6 @@ npm-debug.log
 .jshintrc
 .vs/
 test/unit/three.*.unit.js
+build/three.module.js
```
Why do you modify `.gitignore`?

anhr replied:
I have dicarded changes in .gitignore

Mugen87 replied:
Um, it seems the file is still modified...

anhr committed:
Remove SpriteText.js

I have removed SpriteText.js because this file from my future pull
request. I included this file in mistake. Sorry

anhr committed:
remove examples/js/gui/StereoEffect.js, examples/js/gui/gui.js

anhr committed:
code style

I have tested my code style by http://zz85.github.io/mrdoobapproves/

anhr commented:
I have removed examples/js/gui/StereoEffect.js, examples/js/gui/gui.js . See my [remove examples/js/gui/StereoEffect.js, examples/js/gui/gui.js](https://github.com/mrdoob/three.js/pull/15434/commits/1727c90eacbe605bd2a7e669d8aeb99a3ea232d6
)  commit.

I have tested my code style by http://zz85.github.io/mrdoobapproves/ See my [code style](https://github.com/mrdoob/three.js/pull/15434/commits/47db14e27458717be3064a95a777ad005655839e) commit

anhr committed:
discard .gitignore changes

anhr committed:
discard .gitignore changes

anhr committed:
Cosmetic changes

Cosmetic changes according
https://github.com/mrdoob/three.js/wiki/Mr.doob%27s-Code-Style%E2%84%A2
rules

anhr committed:
Cosmetic changes

Cosmetic changes according
https://github.com/mrdoob/three.js/wiki/Mr.doob%27s-Code-Style%E2%84%A2
rules

anhr committed:
optimized code

anhr committed:
Focus control

Default camera focus is depend from far - Camera frustum far plane.

anhr committed:
info

anhr committed:
default camera focus

anhr committed:
cosmetic changes

anhr committed:
Optimized code

anhr committed:
Add stereo effect into webgl_geometry_convex.html

anhr committed:
dat.folderNameAndTitle

Change name and title of the dat.GUI folder

anhr committed:
THREE.StereoEffect.isTrue

anhr committed:
Optimized code

anhr committed:
Cosmetic changes

anhr committed:
Merge remote-tracking branch 'refs/remotes/mrdoob/dev' into dev

anhr committed:
code style

anhr committed:
save settings of each webpage separately

anhr committed:
Optimized code

takahirox commented:
gui, cookie, and language being in `examples/webgl_effects_stereo.html` looks more natural to me. rather than in `StereoEffect.js`. Especially gui setting is in .html in other examples. If we move, `StereoEffect.js` will be simpler. What do you folks think?

anhr commented:
@takahirox Currently I have added StereoEffect into [webgl_geometry_convex.html](https://github.com/mrdoob/three.js/pull/15434/commits/cc663214142bfb8f914b5ce30c3430f7df35b5a1) also. If we  move gui, cookie, and language from StereoEffect.js to examples/webgl_effects_stereo.html then we need move same code into webgl_geometry_convex.html. As result we will have duplicate code for each web page if you want add StereoEffect. 

To avoid duplicate code and StereoEffect.js will be simpler, I have moved gui, cookie, and language from StereoEffect.js to separate files. See [StereoEffects gui folder](https://github.com/mrdoob/three.js/pull/15434/commits/1cda22d4e412c9df440910fc02a917ea78bc5c54).  But @Mugen87 asked me to move it back:

> I don't think it's good that this PR creates a lot of new files (e.g. examples/js/gui/StereoEffect.js, examples/js/gui/gui.js or src/objects/SpriteText.js). I would prefer modifications just to StereoEffect and the respective example.

I agree with @takahirox and agree to move gui, cookie, and language from StereoEffect.js to separate files again.

anhr committed:
Optimized cookie code

Now I save user setting into single object. As result I avoid the
potential for ambiguity of names of settings for different objects. Also
I now I can save different settings for same object by changing of name
of the object

anhr committed:
debug of the saving of user options and other

takahirox commented:
Maybe I'm missing something. What is adding the stereo effect to geometry convex example for?

anhr committed:
error

anhr committed:
Load of options bug

Now values of gui of objects with common options is correct

anhr committed:
Load object bug

anhr committed:
THREE.cookie.getObject bug

anhr committed:
dat.controllerZeroStep

anhr committed:
dat.controllerZeroStep

Solving of dat.gui NumberController Step bug.
See
https://github.com/dataarts/dat.gui/issues/48 for details.

anhr committed:
setValue of dat.controllerZeroStep

anhr committed:
code style

anhr committed:
remember options

remember options only if guiParams.gui is defined

anhr committed:
THREE.cookie.getObject bug

anhr committed:
stereoAspect

anhr committed:
Resolving conflict

anhr committed:
Resolving conflict

anhr committed:
Resolving conflict

anhr committed:
update from mrdoob

anhr committed:
Revert "THREE.cookie.getObject bug"

This reverts commit 95c31f0775de4b62d5f0e593f767a38d1b1223a8.

anhr committed:
AxesHelper

anhr committed:
Merge remote-tracking branch 'refs/remotes/mrdoob/dev' into dev

# Conflicts:
#	examples/js/effects/StereoEffect.js

anhr committed:
cosmetic changes in MathBox.js

anhr committed:
renderer.getSize compatibility

I have modified renderer.getSize  for excluding of warning:
"WebGLRenderer: .getsize() now requires a Vector2 as an argument"

anhr committed:
mathbox

anhr committed:
webgl_math.html

anhr committed:
webgl_raycaster_stereo.html

anhr committed:
esolving conflicts

anhr committed:
code style

anhr committed:
Parsing error: Deleting local variable in strict mode

anhr committed:
Parsing error: Deleting local variable in strict mode

anhr committed:
code style

anhr committed:
remove files

anhr committed:
error  Newline required at end of file but not found  eol-last

mrdoob commented:
Uh oh... This got out of control...
